src/main/java/hudson/plugins/promoted_builds/PromotionProcess.java

```diff
@@ -372,11 +386,20 @@ public Status isMet(AbstractBuild<?,?> build) {
         List<PromotionBadge> badges = new ArrayList<PromotionBadge>();
         for (PromotionCondition cond : conditions) {
             PromotionBadge b = cond.isMet(this, build);
-            if(b==null)
-                return null;
-            badges.add(b);
+            if (this.any_met_condition) {
+                if (b!=null)
```

Comment:
nit: Since you're modifying these two if statements anyway, I think it would be good to go ahead and use braces around the conditional statements.